### DISCUSSION TITLE
Conflict with Nethereum about some dll version conflicts

### DISCUSSION THREAD

**Discussion by defchen** [WalletConnect/WalletConnectUnity#163] [Q&A] [answered]
When I imported the Nethereum latest version. There are serveral dll conflicts.

**Comment by skibitsky** [comment]
Hey,

What Unity version do you use and how exactly do you import Nethereum DLLs?

**Reply by defchen** [reply]
I use Unity 2021.3.29. Tried two methods but failed.
1. Imported Nethereum from Unity PackageManager via disk files of Nethereum.Unity-4.19.1. 
<img width="800" alt="截屏2024-02-21 15 43 58" src="https://github.com/WalletConnect/WalletConnectUnity/assets/1761985/3679c52f-869b-4d44-ad16-122029cd79af">

2. Also tried  to add the Nethereum.Unity-4.19.1 runtime package into the Plugins folder directly.

Errors below: 
<img width="841" alt="截屏2024-02-21 15 45 05" src="https://github.com/WalletConnect/WalletConnectUnity/assets/1761985/6d92a308-101c-4528-8d5a-a4b2dd0aa7f7">

**Reply by skibitsky** [reply, marked as answer]
This happens because WalletConnect and Nethereum use different versions of `BouncyCastle` library. 

The only workaround for this version of Unity is to disable assembly version validation in Player Settings ([API reference](https://docs.unity3d.com/2021.3/Documentation/ScriptReference/PlayerSettings-assemblyVersionValidation.html)).

Unity 2022.2 introduced binding redirects, allowing it to choose the newer `BouncyCastle` DLL at runtime, similar to how msbuild does it. I'd recommend you upgrading to 2022.2 or newer.

**Reply by defchen** [reply]
Thank you very much.
